newlib/libc/sys/vita/io.c

```diff
@@ -132,6 +132,35 @@ int __vita_release_descriptor(int fd)
 	return 0;
 }
 
+int __vita_duplicate_descriptor(int fd)
+{
+	int fd2 = -1;
+
+	sceKernelLockLwMutex(&_newlib_fd_mutex, 1, 0);
+
+	if (!is_fd_valid(fd))
+	{
+		sceKernelUnlockLwMutex(&_newlib_fd_mutex, 1);
```

Comment:
Right, I understand the reasoning behind goto, but for the sake of consistency within the source file it would be wise to handle other mutex unlocks in the same way.